File: MATEMÁTICA/2516/Main.java
import java.util.Scanner;

public class Main {

    static double elapsed(int s, int va, int vb) {
        return (double) s / (va - vb);
    }

    static void formattedPrint(double time) {
        System.out.printf(time < 0 ? "impossivel\n" : "%.2f\n", time);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int s = scanner.nextInt();
            int va = scanner.nextInt();
            int vb = scanner.nextInt();
            formattedPrint(elapsed(s, va, vb));
        }
        scanner.close();
    }
}

File: MATEMÁTICA/2516/main.rb
# frozen_string_literal:true

def elapsed(s, va, vb)
  s / (va - vb)
end

def formatted_print(time)
  puts time.negative? ? 'impossivel' : format('%.2f', time)
end

while line = gets
  s, va, vb = line.split.map(&:to_f)
  formatted_print(elapsed(s, va, vb))
end
